### ISSUE TITLE
PolymorphicModel fails if swapped with another Model in Django 1.10

### ISSUE THREAD

**Issue by GheloAce** [django-polymorphic/django-polymorphic#270]
I developed a reusable app called `alpha` that provides the `Alpha` model (that is expected to get inherited by another model `AnotherAlpha`) using `django-polymorphic` and `swapper`.  Then created the `beta` app that inherits `alpha`'s `AlphaAbstract` so that users can decide to use a different Model instead of the default.  After swapping to `beta`'s implementation of `Alpha` i get this error:

```
  [...]
  File "/home/ghelo/PycharmProjects/capricorn/PolySwapper/alpha/models.py", line 10, in <module>
    class Alpha(AlphaAbstract):
  File "/home/ghelo/PycharmProjects/capricorn/lib/python3.6/site-packages/polymorphic/base.py", line 99, in __new__
    self.validate_model_manager(new_class.objects, model_name, 'objects')
  File "/home/ghelo/PycharmProjects/capricorn/lib/python3.6/site-packages/django/db/models/manager.py", line 198, in __get__
    cls._meta.swapped,

AttributeError: Manager isn't available; 'alpha.Alpha' has been swapped for 'beta.Beta'
```

---------------------
## Replication

**1.** Run these commands:
```bash
pip install Django
django-admin startproject PolySwapper
cd PolySwapper
django-admin startapp alpha
django-admin startapp beta
pip install django-polymorphic swapper
```
As of reporting this issue the package versions are:
```
Django==1.10.6
swapper==1.0.0
django-polymorphic==1.1
```
**2.** Create models Alpha and Beta
```python
# alpha/models.py
import swapper
from polymorphic.models import PolymorphicModel
class AlphaAbstract(PolymorphicModel):
    class Meta:
        abstract = True
class Alpha(AlphaAbstract):
    class Meta:
        swappable = swapper.swappable_setting('alpha', "Alpha")
class AnotherAlpha(Alpha):
    pass

# beta/models.py
from alpha.models import AlphaAbstract
class Beta(AlphaAbstract):
    pass
```
**3.** Run `./manage.py makemigrations`
**4.** Add `alpha` and `beta` to `settings.py`:
```python
INSTALLED_APPS = [
    ...
    'alpha',
    'beta',
]
```
**5.** Append the following to `settings.py` like so:
```python
ALPHA_ALPHA_MODEL = "beta.Beta"
```

**6.** Run `./manage.py shell`  (or any command) and you'll get the error:
```
AttributeError: Manager isn't available; 'alpha.Alpha' has been swapped for 'beta.Beta'
```

## Note on Django 1.9

I tried replicating this issue using `Django~1.9` and i get a different error, but i think this is a bug with Django:
```
alpha.AnotherAlpha.alpha_ptr: (fields.E301) Field defines a relation with the model 'alpha.Alpha', which has been swapped out.
	HINT: Update the relation to point at 'settings.ALPHA_ALPHA_MODEL'.
```

This can be fixed by explicitly declaring the parent_link, like so:
```python
class AnotherAlpha(Alpha):
    alpha_ptr = models.OneToOneField(
        auto_created=True, on_delete=CASCADE,
        parent_link=True, primary_key=True, serialize=False,
        to=swapper.get_model_name('alpha', "Alpha"))
```

After that the issue never gets raised.

**Comment by skirsdeda** [comment]
@vdboor any chance you could do 1.3.1 with only #307 merged in on top of 1.3? For example in django-filer, some tests are turned off (using Django 1.10) only because of this issue. Such release would help a lot of people stuck with Django 1.10 for one reason or another.

**Comment by yakky** [comment]
Agree
@vdboor, do you think it's feasible?

**Comment by vdboor** [comment]
@yakky @skirsdeda yes, I've released django-polymorphic v1.3.1 with the fix

**Comment by yakky** [comment]
Thanks @vdboor !!
